topic_dynamics/parsing.py

```diff
@@ -223,56 +250,46 @@ def calculate_diffs(slices_tokens_dir: str, output_dir: str, name: str, dates: L
             previous_version = {}
             with open(os.path.abspath(os.path.join(slices_tokens_dir, str(date - 1) + ".txt"))) as fin:
                 for line in fin:
-                    previous_version[line.rstrip().split(";")[1]] = line.rstrip().split(";")[2]
+                    token_line = TokenLine(*line.split(";"))
+                    previous_version[token_line.address] = read_tokens_counter(token_line.tokens)
             current_version = []
             with open(os.path.abspath(os.path.join(slices_tokens_dir, str(date) + ".txt"))) as fin:
                 for line in fin:
                     # Iterate over files in the "current" version
-                    current_version.append(line.rstrip().split(";")[1])
-                    address = line.rstrip().split(";")[1]
-                    tokens = {}
-                    for token in line.rstrip().split(";")[2].split(","):
-                        tokens[token.split(":")[0]] = int(token.split(":")[1])
-                    tokens = Counter(tokens)
-                    old_address = address.replace(dates[date - 1].strftime("%Y-%m-%d"),
-                                                  dates[date - 2].strftime("%Y-%m-%d"), 1)
+                    token_line = TokenLine(*line.split(";"))
+                    current_version.append(token_line.address)
+                    tokens = read_tokens_counter(token_line.tokens)
+                    old_address = token_line.address.replace(dates[date - 1].strftime("%Y-%m-%d"),
+                                                             dates[date - 2].strftime("%Y-%m-%d"), 1)
                     # Check if the file with this name existed in the previous version
                     if old_address in previous_version.keys():
-                        old_tokens = {}
-                        for token in previous_version[old_address].split(","):
-                            old_tokens[token.split(":")[0]] = int(token.split(":")[1])
-                        old_tokens = Counter(old_tokens)
-                        # Calcualate which tokens have been added and removed between versions
+                        old_tokens = previous_version[old_address]
+                        # Calculate which tokens have been added and removed between versions
                         created_tokens = sorted((tokens - old_tokens).items(), key=itemgetter(1), reverse=True)
```

Comment:
```suggestion
                        changed_tokens = [(token, tokens[token] - old_tokens[token]) for token in tokens if tokens[token] != old_tokens[token]]
```
it can be done in 1 pass